phparkitect.php

```diff
@@ -14,27 +14,94 @@
 
 return static function (Config $config): void
 {
-    $srcClassSet = ClassSet::fromDir(__DIR__.'/src');
+    $specClassSet = ClassSet::fromDir(__DIR__ . '/src/Sylius/{Behat,Component/*,Bundle/*}/spec');
 
-    $rules = [];
+    $config->add(
+        $specClassSet,
+        Rule::allClasses()
+            ->that(new Extend(ObjectBehavior::class))
+            ->should(new HaveNameMatching('*Spec'))
+            ->because('This is a convention from PHPSpec')
+        ,
+        Rule::allClasses()
+            ->that(new Extend(ObjectBehavior::class))
+            ->should(new IsFinal())
+            ->because('Specifications should not be extendable')
+        ,
+    );
 
-    $rules[] = Rule::allClasses()
-        ->that(new ResideInOneOfTheseNamespaces('Sylius\Component'))
-        ->should(new NotDependsOnTheseNamespaces('Sylius\Bundle'))
-        ->because('Sylius components should be stand-alone')
-    ;
+    $testsClassSet = ClassSet::fromDir(__DIR__ . '{/tests,/src/Sylius/Bundle/*/Tests}');
 
-    $rules[] = Rule::allClasses()
-        ->that(new Extend(ObjectBehavior::class))
-        ->should(new HaveNameMatching('*Spec'))
-        ->because('This is a convention from PHPSpec')
-    ;
+    $config->add(
```

Comment:
I think it should be possible to add case as for spec:
`->should(new HaveNameMatching('*Test'))` for classes in `/Tests`, which extend `TestCase` or something similar

Reply:
The tool is frankly pretty dump, the first thing I wanted to do is put `isFinal` on everything that just extends `TestCase` and it works, but **only** for `TestCase`, any other (Kernel, JsonApi, AbstractWeb) is ignored, so there would need to be `n` rules depending on the extension point and it'd need to be kept updated when new ones pop up.
Same with the case you mentioned, they'll either be ignored or fail, IMO it's more hassle than it's worth, at least at the start.
Also out `TestCase`s would fail that constraint since they end w/ `Case` 💃